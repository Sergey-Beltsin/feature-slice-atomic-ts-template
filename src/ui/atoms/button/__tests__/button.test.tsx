import ReactDOM from 'react-dom';
import {Button} from "../index";

import { render, cleanup } from "@testing-library/react";
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

afterEach(cleanup);

it('render without crashing', function () {
  const div = document.createElement('div');
  ReactDOM.render(<Button onClick={() => console.log('rendered')} label='simple button'>Hi</Button>, div);
});

it('render button should be correctly', function () {
  const {getByTestId} = render(<Button onClick={() => console.log('rendered')} label='simple button'>I'm test button</Button>);
  expect(getByTestId('button')).toHaveTextContent('I\'m test button');
});

it('matches snapshot', function () {
  const tree = renderer.create(<Button onClick={() => console.log('renderer')} label='simple button'>Hi, I'm test button</Button>).toJSON();
  expect(tree).toMatchSnapshot();
});
