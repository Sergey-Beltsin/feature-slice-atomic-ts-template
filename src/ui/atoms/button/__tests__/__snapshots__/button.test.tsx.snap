// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  aria-label="simple button"
  className="button"
  data-testid="button"
  onClick={[Function]}
>
  Hi, I'm test button
</button>
`;
